### DISCUSSION TITLE
Images are missing in Media Library after some time

### DISCUSSION THREAD

**Discussion by mvskiren** [strapi/strapi#6830] [Legacy Topics]
Deployed Strapi on Heroku. When I tried to add Images in media Library images are showing only for some time, but after some time images are missing. Does anyone come across this issue?
<img width="1440" alt="Screenshot 2020-06-27 at 3 50 35 PM" src="https://user-images.githubusercontent.com/19667191/85920051-fb0c2c80-b88d-11ea-82ae-c3862e290913.png">

**Comment by MattieBelt** [comment]
Hi @mvskiren, check this FAQ: https://strapi.io/documentation/v3.x/getting-started/troubleshooting.html#why-is-my-app-s-database-and-uploads-resetting-on-heroku

**Comment by jueschus** [comment]
@mvskiren: was this your problem?
@MattieBelt: i have the same issue, but i'm connected to mongodb atlas.
i can see a "upload_file" collection in my database with 5 entries (that's correct). the collection size seems that there are only references stored.. am i missing something in my configuration? where should the "real" media be stored? upload works fine, but after a dyno restart i see only references in my strapi admin.

**Comment by MattieBelt** [comment]
Hey @jueschus, your images are by default stored locally and because: 
> Heroku's file system is ephemeral meaning that each time a dyno (container) is reset all filesystem changes are lost. And since both SQLite and local uploads are stored on the filesystem, any changes made to these since the last dyno reset will be deleted. 

So yes when you restart your dyno you will only have references, the local files are gone.
You should use any strapi-upload-provider, checkout the docs and a npm list here: https://strapi.io/documentation/v3.x/plugins/upload.html#using-a-provider

**Reply by jueschus** [reply]
Thanks a lot!
I found a plugin for storing your files directly in your mongodb, so no need for an addtion to my current setup:
https://github.com/AHgPuK/strapi-mongodb-files
